app/data_models/answer.py

```diff
@@ -1,28 +1,80 @@
 from __future__ import annotations
 
 from dataclasses import asdict, dataclass, field
-from typing import Dict, List, Mapping, Optional, Union
+from decimal import Decimal
+from typing import Optional, Union, overload
+
+from markupsafe import Markup, escape
+
+DictAnswer = dict[str, Union[int, str]]
+ListAnswer = list[str]
+DictAnswerEscaped = dict[str, Union[int, Markup]]
+ListAnswerEscaped = list[Markup]
+
+AnswerValueTypes = Union[str, int, Decimal, DictAnswer, ListAnswer]
+AnswerValueEscapedTypes = Union[
+    Markup,
+    int,
+    Decimal,
+    DictAnswerEscaped,
+    ListAnswerEscaped,
+]
 
 
 @dataclass
 class Answer:
     answer_id: str
-    value: Union[str, int, float, List, Mapping]
+    value: AnswerValueTypes
     list_item_id: Optional[str] = field(default=None)
 
     @classmethod
-    def from_dict(cls, answer_dict: Dict) -> Answer:
+    def from_dict(cls, answer_dict: dict) -> Answer:
         return cls(
             answer_id=answer_dict["answer_id"],
             value=answer_dict["value"],
             list_item_id=answer_dict.get("list_item_id"),
         )
 
-    def for_json(self) -> Dict:
+    def for_json(self) -> dict:
         output = self.to_dict()
         if not self.list_item_id:
             del output["list_item_id"]
         return output
 
-    def to_dict(self) -> Dict:
+    def to_dict(self) -> dict:
         return asdict(self)
+
+
+@overload
+def escape_answer_value(value: ListAnswer) -> ListAnswerEscaped:
+    ...  # pragma: no cover
```

Comment:
This is to improve type annotations and make MyPy understand how argument types link to return types. You can read more about it at https://docs.python.org/3/library/typing.html#typing.overload